cli/storage_s3.go

```diff
@@ -25,9 +28,30 @@ func (c *storageS3Flags) setup(_ storageProviderServices, cmd *kingpin.CmdClause
 	cmd.Flag("disable-tls-verification", "Disable TLS (HTTPS) certificate verification").BoolVar(&c.s3options.DoNotVerifyTLS)
 	cmd.Flag("max-download-speed", "Limit the download speed.").PlaceHolder("BYTES_PER_SEC").IntVar(&c.s3options.MaxDownloadSpeedBytesPerSecond)
 	cmd.Flag("max-upload-speed", "Limit the upload speed.").PlaceHolder("BYTES_PER_SEC").IntVar(&c.s3options.MaxUploadSpeedBytesPerSecond)
+
+	var pointInTimeStr string
+
+	pitPreAction := func(pc *kingpin.ParseContext) error {
+		if pointInTimeStr != "" {
+			t, err := dateparse.ParseStrict(pointInTimeStr)
+			if err != nil {
+				return errors.Wrap(err, "invalid point-in-time argument")
+			}
+
+			c.s3options.PointInTime = &t
+		}
+
+		return nil
+	}
+
+	cmd.Flag("point-in-time", "Use a point-in-time view of the storage repository when supported").PlaceHolder(time.RFC3339).PreAction(pitPreAction).StringVar(&pointInTimeStr)
 }
 
 func (c *storageS3Flags) connect(ctx context.Context, isNew bool) (blob.Storage, error) {
+	if isNew && c.s3options.PointInTime != nil && !c.s3options.PointInTime.IsZero() {
+		return nil, errors.New("Cannot specify a 'point-in-time' option when creating a repository")
```

Comment:
Looks like `connect` can also be called from `repo sync-to` and `repo repair` commands. Will that pose an issue if combined with a `--point-in-time` flag?

I can imagine `sync-to` might work, but `repo-repair` sounds like it wouldn't work with read-only storage. Not too familiar with the details of these features, but I thought I'd bring it up in case it needs to be handled.

Reply:
`repo-repair` would not work and for the most part this is no different than specifying `--readonly` during connect. Perhaps the `repo-repair` would fail earlier and with a more useful message in this case. Need to double check. This is a usability nit, however it's not clear (at least to me) that there is an easy way to fix it given how the CLI flags are structured now.

As for `sync-to`, whether the behavior is the desired one or not depends mostly on (a) the state of the destination; and (b) whether the `--delete` flag is specified.
There is a particular scenario where performing a point-in-time repo sync is extremely useful. Suppose you want to create a copy of the repo at a particular point in time say because (a) it is known to be in a good state; and (b) the original repo is now in read only and kept around for forensics analysis. Assuming the target repo is empty, then by specifying a `--point-in-time` flag, the user would get the desired copy of the repo in the destination.